assert/error.go

```diff
@@ -19,3 +22,13 @@ func Error(t *testing.T, err error, details ...interface{}) {
 		t.Fatalf("expected error, got nil.%s", errordetails(details...))
 	}
 }
+
+// IsError will call Fatal if the given error does not match the want error.
+// It uses the errors.Is() function to check if the error wraps the wanted error.
+func IsError(t *testing.T, want, got error, details ...interface{}) {
+	t.Helper()
+
+	if !errors.Is(got, want) {
+		t.Fatalf("error[%s] does not match wanted[%s].%s", got, want, details)
```

Comment:
Hmm details is a ...interface{} but it is printed just as a string, that will not look good if I pass a lot of parameters on the details. Don't we want the details to be a possibly formatted message ? (hence the ...interface{})

So I could use like:

```go
assert.IsError(t, want, got, "more formatted details: %v", value)
```

If that is the case, we need to fix the function, if that is not the case...we should not use "...interface{}" as details.

```suggestion
		t.Fatalf("got [%s] but wanted [%s]: %s", got, want, errordetails(details...))
```